packages/cashier/src/Components/percentage-selector/percentage-selector.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text } from '@deriv/components';
+import { Localize } from '@deriv/translations';
+
+const PercentageSelector = ({ amount, currency, getCalculatedAmount }) => {
+    const [percentage, setPercentage] = React.useState('0');
+
+    const calculateAmount = (e, percent) => {
+        setPercentage(percent);
+
+        for (let i = 1; i <= 4; i++) {
+            if (i <= e.target.id) {
+                document.getElementById(i).style.backgroundColor = 'blue'; // TODO: Change color when design is updated
+            } else {
+                document.getElementById(i).style.backgroundColor = '#f2f3f4';
+            }
+        }
+        getCalculatedAmount(amount * (percentage / 100));
+    };
+
+    return (
+        <React.Fragment>
+            <div className='percentage-selector'>
+                <div className='percentage-selector__block-container'>
```

Comment:
Since we have same blocks of code which repeated 4 times, can we have a function for them instead?

Reply:
The wrapping div might be the same but the contents are different (id, onClick, percentage value) and instead of sending the components to the function, it's better to repeat a wrapping div.

Reply:
You can define an array for those values and pass them as an array to the component. Since the wrapper and all the class names are the same, it's better to create a component function.

Reply:
Sure.